lib/src/tar/import.rs

```diff
@@ -115,11 +115,11 @@ fn entry_to_variant<R: std::io::Read, T: StaticVariantType>(
     Ok(v.normal_form())
 }
 
-impl<'a> Importer<'a> {
-    fn new(repo: &'a ostree::Repo) -> Self {
+impl Importer {
+    fn new(repo: &ostree::Repo) -> Self {
```

Comment:
I do wonder whether it would be better to take full ownership of the `ostree::Repo` argument from the caller here, and give it back through a new `finish()` method when importing is over. It would also be more in line with usual Rust rules, as the immutable borrow here is a bit of lie (importing steps do mutate the repo). How are your feelings on that?

Reply:
Right; gtk-rs made a core decision that GObjects always have interior mutability.  Because it's really the 90% case.  And it does indeed mean that one can get memory corruption if invoking methods on a gobject from multiple threads if those objects/methods aren't explicitly threadsafe internally.

So I think trying to do such a dance wouldn't really help much.